### PR TITLE
[azopenai] Updating to use KeyCredential

### LINKED ISSUE #21529: [azeventgrid,azopenai] Use the upcoming azcore KeyCredential type
Depends on #20541

Once this type is in we can switch over azeventgrid's two clients (Client, and publisher.Client) as well as azopenai's Clients to use it.

### PR DESCRIPTION
**Pull request by richardpark-msft**
Fixes #21529

### CODE REVIEW

File: sdk/ai/azopenai/custom_client.go [hunk 1]
```diff
@@ -108,22 +115,19 @@ func NewClientForOpenAI(endpoint string, credential KeyCredential, options *Clie
...
+	return &openAIPolicy{}
 }
 
 // Do returns a function which adapts a request to target OpenAI.
 // Specifically, it removes the api-version query parameter.
 func (b *openAIPolicy) Do(req *policy.Request) (*http.Response, error) {
 	q := req.Raw().URL.Query()
 	q.Del("api-version")
-	req.Raw().Header.Set("authorization", "Bearer "+b.cred.apiKey)
+	// req.Raw().Header.Set("authorization", "Bearer "+b.cred.apiKey)
```
Comment (jhendrixMSFT) [inline sdk/ai/azopenai/custom_client.go:130]: Remove?

**Review by jhendrixMSFT — APPROVED** [review]

File: sdk/ai/azopenai/custom_client.go (hunk 1 above)
Comment (richardpark-msft) [reply sdk/ai/azopenai/custom_client.go:130]: Yes!

**Comment by richardpark-msft** [comment]
(ignore that check-enforcer override, wrong PR)

**Pull request merged**